33-android-create-composable-views

viamgr opened the pull request:
Added language screen composables #33

https://github.com/user-attachments/assets/ef423d58-b48a-455c-b68f-df3ad5f3da6a

viamgr committed:
#33 Updated UI dependencies and Ktlint version

- Changed `androidx.ui.tooling.preview` to `debugImplementation`
- Added `androidx.ui.tooling` to `debugImplementation`
- Updated ktlint version to 12.1.2
- Added `uiTooling` version as 1.7.8

viamgr committed:
#33 Added `LanguageSelectionUiModel` data class for language selection screen.

viamgr committed:
#33 feat: Implemented the Language Selection Screen UI

- Created the `LanguageSelectionScreen` composable to display language options.
- Added `LanguageDropdown` for language selection.
- Added `SubmitAndContinueButton` to continue after selection.
- Added `WhatLanguageToLearn` and `SelectLanguageText` for screen labels.
- Added preview for all composables.

viamgr committed:
#33 Updated `LanguageSelectionUiModel` to include `languagesToLearnEnabled` and used it to enable/disable `languagesToLearn` selection in the UI.

viamgr committed:
#33 Removed unused dependencies from `libs.versions.toml` and simplified `LanguageSelectionScreen` code.

- Removed `coroutine`, `junit5`, `kotest`, `kotest-extensions-roboelectric`, `ktor`, `room`, and `ktlint` dependencies from `libs.versions.toml`.
- Removed unused `languagesToLearnEnabled` parameter from `LanguageSelectionScreen`.

mmbehroozfar commented on gradle/libs.versions.toml:
```diff
@@ -36,48 +36,48 @@ androidx-ui-tooling-preview = { group = "androidx.compose.ui", name = "ui-toolin
+#junit5-engine = { group = "org.junit.jupiter", name = "junit-jupiter-engine", version.ref = "junit5" }
+#serialization-json = { group = "org.jetbrains.kotlinx", name = "kotlinx-serialization-json", version = "1.0.1" }
+#
+##Coroutines
+#coroutines-android = { group = "org.jetbrains.kotlinx", name = "kotlinx-coroutines-android", version.ref = "coroutine" }
+#coroutines-core = { group = "org.jetbrains.kotlinx", name = "kotlinx-coroutines-core", version.ref = "coroutine" }
+#coroutines-test = { group = "org.jetbrains.kotlinx", name = "kotlinx-coroutines-test", version.ref = "coroutine" }
```
Do they cause any issues?

viamgr replied:
Yes, whenever I want to push, AS warns me about these unused items. I think we can uncomment them when it´s needed

mmbehroozfar commented on feature/main/ui/src/main/kotlin/com/epiclabs/thinktok/main/ui/languageselection/SubmitAndContinueButton.kt:
```diff
@@ -0,0 +1,24 @@
+
+@Composable
+internal fun SubmitAndContinueButton(
+    modifier: Modifier = Modifier,
+    onClick: () -> Unit = {},
+    buttonText: String,
```
Modifier should be first optional argument after required ones.

```suggestion
    buttonText: String,
    modifier: Modifier = Modifier,
    onClick: () -> Unit = {},
```

viamgr committed:
Refactor: Moved `buttonText` parameter before `modifier` in `SubmitAndContinueButton`

- Reordered the parameters of the `SubmitAndContinueButton` composable function, placing `buttonText` before `modifier`.

viamgr committed:
Merge branch 'develop' of https://github.com/epic-labs-org/ThinkTokAndroid into 33-android-create-composable-views

# Conflicts:
#	gradle/libs.versions.toml

viamgr committed:
Refactor: Removed empty lines in SplashConstant.kt.

mmbehroozfar commented on feature/main/ui/src/main/kotlin/com/epiclabs/thinktok/main/ui/languageselection/LanguageDropdown.kt:
```diff
@@ -0,0 +1,72 @@
+    placeholder: String,
+    onLanguageSelected: (String) -> Unit = {},
+) {
+    val expanded = remember { mutableStateOf(false) }
+    val selectedLanguage = remember { mutableStateOf(placeholder) }
```
```suggestion
    var expanded by remember { mutableStateOf(false) }
    var selectedLanguage by remember { mutableStateOf(placeholder) }
```

viamgr committed:
Update feature/main/ui/src/main/kotlin/com/epiclabs/thinktok/main/ui/languageselection/LanguageDropdown.kt

Co-authored-by: Mohammad Mahdi Behroozfar <mmbehroozfar@gmail.com>

viamgr committed:
Refactored language selection dropdown
